### PR TITLE
Add `Divider` component and an example

### PR DESCRIPTION
**Pull request by ericvalcik**
This PR adds the `Divider` component and some examples for it.

The implementation can seem a little bit shaky, especially the fact that we have to define `dividerWidth` for the `Divider` if we want to show text, but I was trying to get it to look the same on most of the browsers, and we accomplish that.

I considered dividing the component into two components (`Divider` and `DividerWithText`), but finally, I decided not to split the component into two. I think it works best the way it is right now

What do you guys think? @iamhectorsosa @jvorcak

### CODE REVIEW

File: src/mailingui/components/divider/Divider.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  const sectionBorder = {
+    width: `${dividerWidth ? Math.round(dividerWidth * 0.4) : 450}px`,
+  };
+
+  const sectionCenter: CSSProperties = {
+    width: `${dividerWidth ? Math.round(dividerWidth * 0.2) : 150}px`,
+    textAlign: "center",
+    fontSize: "14px",
+  };
```
Comment (iamhectorsosa) [inline src/mailingui/components/divider/Divider.tsx:31]: When you use text, you build three horizontal sections based on what I read here. It is composed of two lateral sections that are 450 px width and a central section is 150 px width, this adds to 1050 px. Where did you come up with this number? How did it work with testing with the default value? All of the examples are using 580 px width as width, maybe we should try to aim for a default that sits around this number?

**Review by iamhectorsosa — DISMISSED** [review]
Looking great! Just have second thoughts on the sections logic.

File: src/mailingui/components/divider/Divider.tsx (hunk 1 above)
Comment (ericvalcik) [reply src/mailingui/components/divider/Divider.tsx:31]: This is a good catch, for the default I wanted to for 600px and divide the sections in 37.5% - 25% - 37.5%. I took inspiration in [this example](https://demo.react.email/preview/twitch-reset-password?view=desktop) from react.email, but I forgot to divide the 450px by two.

I'll fix that. Surprisingly though, the tests went fine for all clients

**Comment by ericvalcik** [comment]
> I expect that the divider would have a smaller base width that my email and it will still render centered, no?

The Divider unfortunately does not render centered if it has a smaller width than the email width.

What I instead did, I made one example smaller in width, and passed that width to the Divider's `baseWidth` as well so we can show more clearly how that prop can be used.

**Review by iamhectorsosa — APPROVED** [review]

**Pull request merged**